packages/grafana-alerting/scripts/codegen.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * This script will generate TypeScript type definitions and a RTKQ client for the alerting k8s APIs.
+ * It downloads the OpenAPI schema from a running Grafana instance and generates the types.
+ *
+ * Run `yarn run codegen` from the "grafana-alerting" package to invoke this script.
+ */
+import { type ConfigFile } from '@rtk-query/codegen-openapi';
+import { resolve } from 'node:path';
+
+// make sure to run the Grafana server before running this script since it will download the OpenAPI schema from the server
+const OPENAPI_SCHEMA_LOCATION = resolve(
+  '../../../pkg/tests/apis/openapi_snapshots/notifications.alerting.grafana.app-v0alpha1.json'
+);
+
+export default {
+  exportName: 'alertingAPI',
+  schemaFile: OPENAPI_SCHEMA_LOCATION,
+  apiFile: '../src/grafana/api.ts',
+  outputFile: resolve('../src/grafana/api.gen.ts'),
+  tag: true,
+} satisfies ConfigFile;
```

Comment:
Suggestion: We could/should probably explicitly list the endpoints that we're using, so the generated client is smaller/its clearer what we depend on from the OpenAPI spec. We could add them as we require them and add new components?

Would just be `filterEndpoints: ['listReceiver']`

Reply:
I'm not entirely sold on that – I imagine some consumers of the package might want to go ahead and use the RTKQ API in whatever way to want, as long as the APIs are there this allows us (and consumers) a way to build bespoke components without having to go through the rather formal process of creating a PR to export / generate more endpoints.

Reply:
Sure, fair enough. We can export the entire API from this package then I guess 👀

Reply:
Yep – this one is exported from `@grafana/alerting/internal`